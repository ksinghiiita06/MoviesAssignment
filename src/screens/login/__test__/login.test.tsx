import React from 'react';
import Login from '../index';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {MenuProvider} from 'react-native-popup-menu';
import {render} from '@testing-library/react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';

describe('Login renders correctly', () => {
  const initialState = {appReducer: {languageCode: 'ar'}};
  const mockStore = configureStore();
  const store = mockStore(initialState);
  let wrapper: any;
  beforeEach(() => {
    wrapper = render(
      <Provider store={store}>
        <MenuProvider>
          <SafeAreaProvider>
            <Login />
          </SafeAreaProvider>
        </MenuProvider>
      </Provider>,
    );
  });
  it('Login renders correctly ', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('Login renders correctly ', () => {
    // fireEvent.changeText(screen.getByTestId('emailInput'), 'kul@gmail.com');
    // fireEvent(screen.getByTestId('pwdInput'), 'onChangeText', 'Passw0rd');
  });
});
