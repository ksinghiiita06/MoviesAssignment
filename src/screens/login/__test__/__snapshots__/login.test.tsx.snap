// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly Login renders correctly  1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    />
  </View>
  <RCTSafeAreaView
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
